pkg/apis/extensions/validation/validation.go

```diff
@@ -279,7 +282,7 @@ func ValidateDeploymentStrategy(strategy *extensions.DeploymentStrategy, fldPath
 		if strategy.RollingUpdate == nil {
 			allErrs = append(allErrs, field.Required(fldPath.Child("rollingUpdate"), "this should be defaulted and never be nil"))
 		} else {
-			allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, fldPath.Child("rollingUpdate"))...)
+			allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, replicas, fldPath.Child("rollingUpdate"))...)
```

Comment:
Maybe we can directly collapse maxUnavailable to spec.replicas here, or somewhere when maxUnavailable is used. As the check of maxUnavailable is not a validation and will not return err, just set its default value when it is larger than spec.replicas.